posthog/hogql/printer.py

```diff
@@ -278,12 +271,11 @@ def visit_join_expr(self, node: ast.JoinExpr) -> JoinExprResponse:
             if self.dialect == "clickhouse":
                 sql = table_type.table.to_printed_clickhouse(self.context)
 
-                # Always put S3 Tables in a CTE so joins can work when queried
-                if isinstance(table_type.table, S3Table):
-                    cte = f"{self._print_identifier(node.alias)} AS (SELECT * FROM {sql})"
-
-                    # The table is captured in a CTE so just print the table name in the final select
-                    sql = self._print_identifier(node.alias or table_type.table.name)
+                # Edge case. If we are joining an s3 table, we must wrap it in a subquery for the join to work
```

Comment:
Is there a ClickHouse issue for this BTW? It seems quite mysterious

Reply:
My guess is that it's a gotcha with remote queries but yeah I'll open one